extmod/modbluetooth_nimble.c

```diff
@@ -587,6 +601,19 @@ int mp_bluetooth_gatts_indicate(uint16_t conn_handle, uint16_t value_handle) {
     return ble_hs_err_to_errno(ble_gattc_indicate(conn_handle, value_handle));
 }
 
+int mp_bluetooth_gatts_set_buffer(uint16_t value_handle, size_t len, bool append) {
+    mp_map_elem_t *elem = mp_map_lookup(MP_STATE_PORT(bluetooth_nimble_root_pointers)->gatts_db, MP_OBJ_NEW_SMALL_INT(value_handle), MP_MAP_LOOKUP);
+    if (!elem) {
+        return MP_EINVAL;
+    }
+    gatts_db_entry_t *entry = MP_OBJ_TO_PTR(elem->value);
+    entry->data = m_new(uint8_t, len);
```

Comment:
Might be better to use `m_renew(uint8_t, entry->data, entry->data_alloc, len)`, to reuseresize memory where possible

Reply:
I considered using m_renew (and truncating data_len to the new data_alloc) to preserve existing data, but scrapped that because the intention is that it's only called once at registration.

But just for the purpose of possible reusing the allocation, makes sense. Done.